syntax = "proto3";

message SpeFrame {
  repeated float stokesi = 1;
  repeated float stokesv = 2;
  repeated double time = 3;
  repeated double freq = 4;
}

message ImgFrame {
  repeated float stokesi = 1;
  repeated float stokesv = 2;
  repeated float sunx = 3;
  repeated float suny = 4;
  int32 time = 5;
  float freq = 6;
}


message OpenImgFileAck {
  ImgFrame frame = 1;
  map<string, string> header0 = 2;
  map<string, string> header1 = 3;
  int32 index = 4;
}

message ImgAppAck{
 ImgFrame frame = 1;
 int32 index = 2;
}

message FlowCalAck {
 map<int32, double> stokesi = 1;
 map<int32, double> stokesv = 2;
}
